localstripe/server.py

```diff
@@ -262,8 +262,12 @@ async def f(request):
             data['source_id'] = request.match_info['source_id']
         if 'subscription_id' in request.match_info:
             data['subscription_id'] = request.match_info['subscription_id']
+        if 'tax_id' in request.match_info:
+            data['tax_id'] = request.match_info['tax_id']
         expand = data.pop('expand', None)
-        return json_response(func(**data)._export(expand=expand))
+        ret = func(**data)
+        return json_response(ret if not hasattr(ret, '_export') else
+                             ret._export(expand=expand))
```

Comment:
(sorry I missed that on first review)

If I remember correctly, the spirit of these `api_update()` / `api_delete()` / `api_extra()` etc. is to only return JSON for types that we expect. Here, it looks like you modify `api_extra()` for your use-case (return direct JSON in a new *delete* case), but not the other functions. Also, if a new function returns something else than a `dict` or `None`, it cannot be serializable by `json_response()`.

Instead of this implementation, I would rather see `_api_delete_tax_id()` return a new `DeletedObject` type (maybe it should inherit from `StripeObject`) to keep things consistent. What do you think?

Reply:
Other (simpler) option: do like in `api_delete()` and allow returning either a `dict` or an object with `_export()` (but only those two):
https://github.com/adrienverge/localstripe/blob/cb071d2f18fa74e1e536e9e27c0733b6c7c69b8e/localstripe/server.py#L242

Reply:
Declaring a new `DeletedObject` type was my initial idea until I saw https://github.com/adrienverge/localstripe/blob/cb071d2f18fa74e1e536e9e27c0733b6c7c69b8e/localstripe/resources.py#L156-L161

:smile: (it's the only place where we're doing this though).
So I did it this way to keep the code similar, but I agree that this `ret if not hasattr(ret, '_export')` is ugly. I'll update both call site to use a `DeletedObject` if you agree!

Reply:
> Other (simpler) option: do like in api_delete() and allow returning either a dict or an object with _export() (but only those two):

Our comments crossed, +1 for this solution that's indeed simpler

Reply:
:+1: 

Both options (`isinstance(ret, dict)` and `DeletedObject`) are good to me, with a slight preference for `DeletedObject` which seems more extendable.

Reply:
Much more satisfying :slightly_smiling_face: